lightning/src/ln/onion_utils.rs

```diff
@@ -2501,6 +2501,187 @@ mod tests {
 		assert_eq!(decrypted_failure.onion_error_code, Some(0x400f));
 	}
 
+	#[test]
+	fn test_trampoline_onion_decryption() {
```

Comment:
This one can also be merged with the previous commit. Again, that would be my pref.